tree.c

```diff
@@ -0,0 +1,206 @@
+#include<stdio.h>
+
+#include<stdlib.h>
+
+#include<ctype.h>
+
+struct tree
+
+{
+
+	int data;
+
+	struct tree *left;
+
+	struct tree *right;
+
+}*head;
+
+void insert(int a)
+
+{
+
+	int count=0;
+
+	struct tree *temp,*ptr,*prev;
+
+	temp=(struct tree *)malloc(sizeof(struct tree));
+
+	temp->data=a;
+
+	temp->left=NULL;
+
+	temp->right=NULL;
+
+	if(head==NULL)
+
+	{
+
+	head=temp;
+
+	}
+
+	else
+
+	{
+
+	ptr=head;
+
+	while(ptr!=NULL)
+
+	{
+   
+ 	prev=ptr;
+
+	if(ptr->left!=NULL)
+
+	{
+ 
+	if(ptr->right==NULL)
+
+	{
+ 	
+	ptr->right=temp;
+	
+	printf("\n%d inserted",a);
+	
+	return;
+	}
+ 
+	else
+       
+	{
+        
+	if(count==0)
+	 
+	{
+	  
+	count=1;
+  	  
+	ptr=ptr->left;
+	 
+	}
+	
+	else
+ 	 
+	{
+	  
+	count=0;
+	  
+	ptr=prev->right;
+	 
+	}
+       
+	}
+	
+}
+
+	else
+
+	{
+
+	ptr->left=temp;
+
+	printf("\n%d inserted",a);
+
+	return;
+
+	}
+	
+}
+
+	}
+
+	printf("\n%d inserted",a);
+
+}
+
+void print(struct tree *p)
+
+{
+	if(p==NULL)return;
+
+	else
+
+	{
+
+	if(p->data!=-1)
+
+	{
+
+	printf("%d ",p->data);
+
+	}
+
+	print(p->left);
+
+	print(p->right);
+
+	}
+
+}
+
+void delete(int a,struct tree *p)
+
+{
+
+	if(p==NULL)
+	return;
+
+	else
+
+	{
+	
+if(p->data==a)
+
+	{
+
+	p->data=-1;
+
+	printf("\n%d deleted",a);
+	return;
+
+	}
+
+	delete(a,p->left);
+
+	delete(a,p->right);
+
+	}
+
+}
+
+int main()
+
+{
+
```

Comment:
Make the interface as generic as possible by taking inputs from stdin. And do test your code for large inputs, for example, 10000 elements.